exercises/state-of-tic-tac-toe/canonical-data.json

```diff
@@ -0,0 +1,369 @@
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XX ",
+              "OOO",
+              "X  "
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "6ef641e9-12ec-44f5-a21c-660ea93907af",
+          "description": "Finished game where O won via row victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XOX",
+              " XX",
+              "OOO"
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "ab145b7b-26a7-426c-ab71-bf418cd07f81",
+          "description": "Finished game where X won via diagonal victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XOO",
+              " X ",
+              "  X"
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "7450caab-08f5-4f03-a74b-99b98c4b7a4b",
+          "description": "Finished game where X won via diagonal victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "O X",
+              "OX ",
+              "X  "
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "c2a652ee-2f93-48aa-a710-a70cd2edce61",
+          "description": "Finished game where O won via diagonal victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "OXX",
+              "OOX",
+              "X O"
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "5b20ceea-494d-4f0c-a986-b99efc163bcf",
+          "description": "Finished game where O won via diagonal victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "  O",
+              " OX",
+              "OXX"
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "035a49b9-dc35-47d3-9d7c-de197161b9d4",
+          "description": "Finished game where X won via a row and a column victory",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XXX",
+              "XOO",
+              "XOO"
+            ]
+          },
+          "expected": "win"
+        },
+        {
+          "uuid": "e5dfdeb0-d2bf-4b5a-b307-e673f69d4a53",
+          "description": "Finished game where X won via two diagonal victories",
+          "property": "gamestate",
+          "input": {
+            "board": [
+              "XOX",
+              "OXO",
+              "XOX"
+            ]
+          },
+          "expected": "win"
+        }
+      ]
+    },
+    {
+      "description": "Drawn games",
+      "cases": [
+        {
+          "uuid": "b42ed767-194c-4364-b36e-efbfb3de8788",
+          "description": "Draw",
```

Comment:
Should the descriptions be unique?  Is this being used for test functionality (function names, etc?) when generated?

Reply:
That's not part of the spec and therefore would be an issue the generators would have to fix.

That said, the descriptions don't describe _why_ the test exists all that well, so if someone wants to improve those, a follow-up PR would be appreciated. The description is mutable and I'd rather get this merged and improved later on according to the usual optimistic merging strategy.